crypto/asn1/a_dup.c

```diff
@@ -48,14 +48,28 @@ void *ASN1_dup(i2d_of_void *i2d, d2i_of_void *d2i, const void *x)
 
 void *ASN1_item_dup(const ASN1_ITEM *it, const void *x)
 {
+    const ASN1_AUX *aux = it->funcs;
+    ASN1_aux_cb *asn1_cb = aux != NULL ? aux->asn1_cb : NULL;
     unsigned char *b = NULL;
     const unsigned char *p;
     long i;
-    void *ret;
+    ASN1_VALUE *ret;
 
     if (x == NULL)
         return NULL;
 
+    /*
+     * TODO remove workaround (resetting asn1_cb) for following issue:
+     * 74080000:error::asn1 encoding routines:asn1_item_embed_d2i:illegal options on item template:..\crypto\asn1\tasn_dec.c:178:Type=X509_NAME_INTERNAL
+     * 74080000:error::asn1 encoding routines:ASN1_item_dup:aux error:..\crypto\asn1\a_dup.c:83:Type=X509_NAME
+    */
+    if (asn1_cb != NULL && strcmp(it->sname, "X509_NAME") == 0)
+        asn1_cb = NULL;
```

Comment:
I can' say I understand what this is all about

Reply:
Well, this hides the strange bug that the cb is set but returns failure. IMO this should not be merged without finding out why the bug happens.

Reply:
@mattcaswell, as mentioned above this is a workaround for a very strange behavior of the `asn1_cb` on some Windoze architectures.

Reply:
I've meanwhile statically restricted the workaround to the Windows platform, where it occurs (only) .